pymeasure/instruments/jobinyvon/jobinyvon270m.py

```diff
@@ -0,0 +1,360 @@
+        """
+        return self._lambda_max - ((self._max_steps - self.gsteps) / self._steps_nm)
+
+    def move_grating_wavelength(self, wavelength: float):
+        """
+        ABSOLUTE positioning of the grating motor in wavelength.
+        """
+        steps = int(self._max_steps - int((self._lambda_max - wavelength) * self._steps_nm))
+        self.move_grating_steps(steps)
+
+    def move_entry_slit_steps(self, nsteps: int):
+        """
+        ABSOLUTE positioning of the entry slit motor in number of steps.
+        """
+        ans = self.write_read(f'k0,0,{nsteps - self.entrysteps}\r'.encode(), nread=1, timeout=10000)
+        time.sleep(0.5)
+        code = self._get_code(ans)
+        if code != 'o':
+            raise IOError(f'Wrong return code from driver, received {code}')
+
+    def get_entry_slit_microns(self):
+        """
+        Reading of the ABSOLUTE position of the entrance slit in micrometres.
+        """
+        return self.entrysteps / self._slit_steps_micron
+
+    def move_entry_slit_microns(self, microns: float):
+        """
+        ABSOLUTE positioning of the entry slit in micrometers.
+        """
+        pos_steps = microns * self._slit_steps_micron
+        pos_steps = int(pos_steps)
+        self.move_entry_slit_steps(pos_steps)
+
+    def move_exit_slit_steps(self, nsteps: int):
+        """
+        ABSOLUTE positioning of the exit slit motor in number of steps.
+        """
+        ans = self.write_read(f'k0,2,{nsteps - self.exitsteps}\r'.encode(), nread=1, timeout=10000)
+        time.sleep(0.5)
+        code = self._get_code(ans)
+        if code != 'o':
+            raise IOError(f'Wrong return code from driver, received {code}')
+
+    def get_exit_slit_microns(self):
+        """
+        Reading of the ABSOLUTE position of the exit slit in micrometres.
+        """
+        return self.exitsteps / self._slit_steps_micron
+
+    def move_exit_slit_microns(self, microns: float):
+        """
+        ABSOLUTE positioning of the exit slit in micrometres.
+        """
+        pos_steps = microns * self._slit_steps_micron
+        pos_steps = int(pos_steps)
+        self.move_exit_slit_steps(pos_steps)
+
+    def check_get_errors(self, error):
+        print(error)
+        pass
+
+    def motor_stop(self):
+        """
+        Stopping of all the motors when this function is called.
+        """
+        ans = self.write_read(b'L')
+        code = self._get_code(ans)
+        if code != 'o':
+            raise IOError(f'Wrong return code from driver, received {code}')
+
+    def motor_busy_check(self):
+        """
+        This function checks if at least one of the motors is busy.
+        """
+        ans = self.write_read(b'E\r').decode()
+        if ans == "oz":
+            "We return False if the motor is not busy."
+            return False
+        if ans == "oq":
+            "We return True if the motor is busy."
+            return True
```

Comment:
Why not a property: `motor_busy`, which will return True or False?
```suggestion
   motor_busy = Instrument.measurement("E\r", values={True: "oq", False: "oz"}, map_values=True)
```

Reply:
Beceause we think the embedded function for writing commands in PyMeasure does not work properly for this instrument, as we have to rely on timeouts to write and read messages. This is why we created a dedicated function called write_read to write and read messages.

Reply:
If you override `ask` with your `write_read` method, you could use it in the regular pymeasure properties!